server/models/singleSourceOfTruthModel.js

```diff
@@ -0,0 +1,39 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const robotMetadataSchema = new Schema({
+    robotId: String,
+    starterId: String
+});
+
+const inputParameterSchema = new Schema({
+    name: String,
+    value: Schema.Types.Mixed,
+    requireUserInput: Boolean
+});
+
+const instructionSchema = new Schema({
+    type: String,
+    name: String,
+    predecessorIds: [mongoose.Types.ObjectId],
+    successorIds: [mongoose.Types.ObjectId],
+    rpaApplication: String,
+    rpaTask: String,
+    rpaParameters: [inputParameterSchema],
+    outputVariable: Schema.Types.Mixed
+});
+
+const markerSchema = new Schema({
+    type: String,
+    name: String,
+    predecessorIds: [mongoose.Types.ObjectId],
+    successorIds: [mongoose.Types.ObjectId]
+});
+
+const SSoT_Schema = new Schema({
+    robotMetadata: robotMetadataSchema,
+    elements: [instructionSchema]
```

Comment:
Elements can also have a markerSchema if I am not mistaken. Can we add that here?